src/main.c

```diff
@@ -1,27 +1,155 @@
-#include "pxbm.h"
-#include <stdio.h>
-#include <stdlib.h>
-
-int main(int argc, char *argv[]) {
-	if(argc < 5) {
-		printf("Error: Missing arguments!\n");
-		printf("./pxbm <filepath> <Red> <Green> <Blue>\n");
-		return(1);
-	}
-	// argv[1] is the filename
-	char *data = read_file(argv[1]);
-
-	if(data) {
-		XBM xptr; 
-		int res = parse_xbm(&xptr, data);
-
-		if(res != 0) {
-			printf("Error: Could not parse xbm data");
-		}
-		// Handle RGB input
-		print_xbm(&xptr, atoi(argv[2]), atoi(argv[3]), atoi(argv[4]));
-	} else {
-		printf("Error: Could not read file. Corrupted or missing?");
-		return(2);
-	}
-}
+#include "pxbm.h"
+#include "netpbm/netpbm.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+
+typedef enum _format
+{
+    FORMAT_NETPBM,
+    FORMAT_XBM,
+} FORMAT;
+
+static FORMAT format = FORMAT_XBM;
+
+static const struct option long_options[] =
+{
+    {"help", no_argument, NULL, 'h'},
+    {"netpbm", no_argument, NULL, 'n'},
+    {"xbm", no_argument, NULL, 'x'},
+    {0, 0, 0, 0}
+};
+
+void usage(int argc, char *argv[])
+{
+    fprintf(stderr,
+            "Usage:\n"
+            "\t%s [options] <filepath> [red] [green] [blue]\n"
+            "Options:\n"
+            "\t-x, --xbm    Print an xbm bitmap (default)\n"
+            "\t-n, --netpbm Print a netpbm bitmap\n"
+            "\t-h, --help   Show this help\n"
+            "Colors:\n"
+            "\tSupply integer arguments in the range 0-255 for\n"
+            "\tthe red, green, and blue channels. Only applies to\n"
```

Comment:
`Only applies to bitmaps that do not support color` should be `Only applies to bitmaps that support color`

Reply:
Nope, only to bitmaps that do not support color. The supplied RGB triple is used as the color. Bitmaps with color support ignore the RGB arguments, as they have their own color embedded in the file.

Reply:
Ohyea.. right. I interpreted that wrong 😂